crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -77,3 +80,51 @@ impl Verified {
         self.inner.boundary
     }
 }
+
+/// Gas parameters associated with a settlement.
+#[derive(Clone, Copy, Debug)]
+pub struct Gas {
+    /// The gas estimate, in gas units, for executing a settlement transaction.
+    pub estimate: eth::Gas,
+    /// The gas limit, in gas units, for a settlement transaction. This is
+    /// computed by adding a buffer to the gas estimate to allow for small
+    /// variations in the actual gas that gets used.
+    pub limit: eth::Gas,
+    /// The maximum fee per unit of gas for a given settlement.
+    pub price: eth::MaxFeePerGas,
+}
+
+impl Gas {
+    /// Computes settlement gas parameters given estimates for gas and gas
+    /// price.
+    pub fn new(estimate: eth::Gas, price: eth::GasPrice) -> Self {
+        // Compute an upper bound for `max_fee_per_gas` for the given
+        // settlement. We multiply a fixed factor of the current base fee per
+        // gas, which is chosen to be the maximum possible increase to the base
+        // fee per gas over 10 blocks.
```

Comment:
I think it makes sense to name the `1.125` base fee scaling constant here and write out the formular how you arrived at the `MAX_FEE_FACTOR`. Maybe something like:
```rust
/// `MAX_FEE_FACTOR` = `MAX_GAS_INCREASE_PER_BLOCK` ^ `SUBMISSION_DEADLINE_IN_BLOCK`
///                  = 1.125 ^ 10
```
Also some link explaining why `1.125` makes sense would be nice.